CatalogAPI/Controllers/CategoriesController.cs

```diff
@@ -0,0 +1,82 @@
+﻿using CatalogAPI.Data;
+using CatalogAPI.Data.Model;
+using CatalogAPI.DTO;
+using Microsoft.AspNetCore.Mvc;
+namespace CatalogAPI.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class CategoriesController : ControllerBase
+    {
+        [HttpGet]
+        public ActionResult GetCategories()
+        {
+            var categoryDTOs = Context.Repository
+                .Select(x => new CategoryDTO()
+                {
+                    Id = x.Id,
+                    Name = x.Name
+                })
+                .ToList();
+            return Ok(categoryDTOs);
+        }
+        [HttpGet("{id}")]
+        public ActionResult GetCategory(int id)
+        {
+            var categoryDTO = Context.Repository
+                .Select(x => new CategoryDTO()
+                {
+                    Id = x.Id,
+                    Name = x.Name
+                })
+                .SingleOrDefault(x => x.Id == id);
+            if (categoryDTO == null)
+            {
+                return NotFound();
+            }
+            return Ok(categoryDTO);
+        }
+        [HttpPost]
+        public ActionResult AddCategory(CategoryDTO categoryDTO)
+        {
+            if (Context.Repository.Any(x => x.Id == categoryDTO.Id))
+            {
+                return BadRequest();
```

Comment:
Richardson maturity level 2  violation. A user is trying to create a resource that already exist

Reply:
While creating I did not get Id from client and I will create new Guid everytime.